src/app/lib/getMerklRewards.ts

```diff
@@ -0,0 +1,25 @@
+import type { MerklRewardsData, StrapiResponse } from '@/types/strapi';
+import { MerklRewardsStrapiApi } from '@/utils/strapi/StrapiApi';
+import { getStrapiApiAccessToken } from 'src/utils/strapi/strapiHelper';
+
+export async function getMerklRewards() {
+  const urlParams = new MerklRewardsStrapiApi();
+  const apiUrl = urlParams.getApiUrl();
+  const accessToken = getStrapiApiAccessToken();
+  const res = await fetch(decodeURIComponent(apiUrl), {
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+    },
+    next: {
+      revalidate: 60 * 5, // revalidate every 5 minutes
+    },
+  });
+
+  if (!res.ok) {
+    throw new Error('Failed to fetch data');
+  }
+
+  const data: StrapiResponse<MerklRewardsData> = await res.json();
+
+  return { data };
```

Comment:
You're doing `{ data: merklRewardsResponse }` then `merklRewardsResponse.data` in the page, return the payload & simplif the code:

```suggestion
  return data.data;
```

Reply:
I'll tackle this in a separate PR in order to align all ways data is returned from a Strapi endpoint 😓